backend/src/apiserver/server/util.go

```diff
@@ -54,13 +55,21 @@ func loadFile(fileReader io.Reader, maxFileLength int) ([]byte, error) {
 }
 
 func isSupportedPipelineFormat(fileName string) bool {
-	return isYamlFile(fileName) || strings.HasSuffix(fileName, ".tar.gz")
+	return isYamlFile(fileName) || isTarballFile(fileName) || isZipFile(fileName)
 }
 
 func isYamlFile(fileName string) bool {
 	return strings.HasSuffix(fileName, ".yaml") || strings.HasSuffix(fileName, ".yml")
 }
 
+func isZipFile(fileName string) bool {
+	return strings.HasSuffix(fileName, ".zip")
+}
+
+func isTarballFile(fileName string) bool {
+	return strings.HasSuffix(fileName, ".tar.gz")
```

Comment:
What about tarballs that aren't compressed? Are those not supported? If we control the generation of the file, why support several types?

Reply:
I can deprecate the tar.gz in a separate change otherwise we might break sample tests. (and I'd prefer do it after a while after .zip is widely used, to avoid back and forth if hub team change minds.)

Reply:
+1. I am fine with the backend supporting multiple formats (zip, tar.gz, yaml) until we become confident that there won't be further changes. 

In the long term: 
- zip: needed. 
- yaml: needed. 
- targ.gz: we might remove it (TBD)